lifecycle-coroutines/src/commonMain/kotlin/com/arkivanov/essenty/lifecycle/coroutines/LifecycleCoroutinesExt.kt

```diff
@@ -0,0 +1,147 @@
+package com.arkivanov.essenty.lifecycle.coroutines
+
+import com.arkivanov.essenty.lifecycle.Lifecycle
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.resume
+
+/**
+ * Repeat invocation [block] every time that passed [state] appears.
+ * Work of passed [block] finished when "opposite" [Lifecycle.State] will appear.
+ *
+ * Note: This function works like a terminal operator and must be called in assembly coroutine.
+ */
+suspend fun Lifecycle.repeatOnEssentyLifecycle(
+    state: Lifecycle.State,
+    block: suspend CoroutineScope.() -> Unit
+) {
+    require(state != Lifecycle.State.INITIALIZED) {
+        "repeatOnEssentyLifecycle cannot start work with the INITIALIZED lifecycle state."
+    }
+
+    if (this.state == Lifecycle.State.DESTROYED) return
```

Comment:
Could you please add curvy braces `{}` for this and all similar cases?

Reply:
Corrected.